PR: https://github.com/odoo/odoo/pull/212679

From: 2713876dbc70d3984e584a9037a2206dcda4e84a
From: Arnold Moyaux
Date: 2025-09-07 15:50:27

Breaking data model changes scores: del:1 + add:0, change matches:
-    _inherit = 'procurement.group'

Total Changes: 33

[REF] stock, *: add reference

The model `procurement.group` is replaced by a more flexible one:
`stock.reference`. The first one was somehow not well understood and a
bit limited by its many2one links to other business model (Sales Order,
Purchase Order, ...). The reference is made of a name and many2many to
all the documents integrated with Inventory. It's purpose has been
shrank to the minimal "make links between object". For instance
opening the manufacturing order created for a MTO sale. The merge of
stock move is delegated to existing parameters (`sale_id`,
created_purchase_line_ids`, ...)

The `mrp.production` needed an extra effort to group them according to
inheritance relationships. This commit add a new object to group
productions together either in a backorder relation or in a child-parent
one. This allow for instance to have a Sale Order linked to the MTO
product of the sold product and the production them selves to be linked
to the SO *and* the potential child MOs.

The procurement group propagation, before set on the Stock Rule, allowing
the stock moves to be merged in an existing picking or not has been
removed and replaced (for the creation of purchase order only) by a new
parameter on `res.partner` allowing the group of Request for Quotation
base on the scheduled date of the potential merge candidate (never, same
week, same month, always). All other merge scenarios should use batches
and waves transfers.

closes odoo/odoo#212679

Task: 4279803
Related: odoo/enterprise#87170
Related: odoo/upgrade#8326
Signed-off-by: Arnold Moyaux (arm) <arm@odoo.com>

================================= pseudo patch: =================================

--- a/addons/product_expiry/models/__init__.py
+++ b/addons/product_expiry/models/__init__.py
@@ -8,3 +8,4 @@ from . import stock_move_line
 from . import stock_move
 from . import stock_picking
 from . import stock_quant
+from . import stock_rule

--- a/addons/product_expiry/models/production_lot.py
+++ b/addons/product_expiry/models/production_lot.py
@@ -105,18 +105,3 @@ class StockLot(models.Model):
         alert_lots.write({
             'product_expiry_reminded': True
         })
-
-
-class ProcurementGroup(models.Model):
-    _inherit = 'procurement.group'
-
-    @api.model
-    def _run_scheduler_tasks(self, use_new_cursor=False, company_id=False):
-        super()._run_scheduler_tasks(use_new_cursor=use_new_cursor, company_id=company_id)
-        self.env['stock.lot']._alert_date_exceeded()
-        if use_new_cursor:
-            self.env['ir.cron']._commit_progress(1)
-
-    @api.model
-    def _get_scheduler_tasks_to_do(self):
-        return super()._get_scheduler_tasks_to_do() + 1

--- a/None
+++ b/addons/product_expiry/models/stock_rule.py
@@ -0,0 +1,17 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, models
+
+
+class StockRule(models.Model):
+    _inherit = 'stock.rule'
+
+    @api.model
+    def _run_scheduler_tasks(self, use_new_cursor=False, company_id=False):
+        super()._run_scheduler_tasks(use_new_cursor=use_new_cursor, company_id=company_id)
+        self.env['stock.lot']._alert_date_exceeded()
+        if use_new_cursor:
+            self.env['ir.cron']._commit_progress(1)
+
+    @api.model
+    def _get_scheduler_tasks_to_do(self):
+        return super()._get_scheduler_tasks_to_do() + 1
